### PR TITLE
Superhero-Search-Project

### PR DESCRIPTION
**Pull request by bhushan354**
In This PR I have used Following things while implementing features in my webApp : 

- Correct GitFlow Used.
- Used React documentation.
- Used React components.
- Used React props.
- Used React Router.
- Connected React and Redux.
- Handled events in a React app.
- Wrote integration tests with a React testing library.
- Used styles in a React app.
- Used React life cycle methods.
- Applied React best practices and language style guides in code.
- Used store, actions and reducers in React.

Thank you ,
Bhuhsan .

### CODE REVIEW

File: README.md
Comment (skyv26) [inline README.md:1]: - [x] Please add the `video` presentation link as per the requirement, your video should not be more than 5 min. You can use the loom for recording and then make a public shareable link. So that we as CR can look into that in order to get a review of what you did and what is special about your project. Kindly stick with the requirements tightly.

Comment (skyv26) [inline README.md:1]: - [x] Kindly always make sure that your `README` file is professional and has everything in a well-structured way as the things provided by Microverse in a well-documented way. Your project description seems irrelevant to the project and needs the proper formatting with a short description of your project.

image

File: LICENSE [hunk 1]
```diff
@@ -1,121 +1,7 @@
...
-### Key Features <a name="key-features"></a>
-
-- When You Search you will see the searched results filtered .
-- Used react-router to naviggate through pages.
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-## 🚀 Live Demo <a name="live-demo"></a>
-
-- [[Live Demo link](https://bhushan354.github.io/Superhero-Search-Project/)]
```
Comment (skyv26) [inline LICENSE:45]: - [ ] I noticed that your live link is not working, rather it shows your actual live web app, it just shows your `README` file, which is totally unprofessional, irrelevant and not acceptable as per the requirement. Kindly deploy your project properly and share the correct deployed link in the `README` file.

image

File: LICENSE [hunk 2]
```diff
@@ -1,121 +1,7 @@
...
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-
-## 🤝 Contributing <a name="contributing"></a>
-
-Contributions, issues, and feature requests are welcome!
-
-Feel free to check the [issues page](https://github.com/bhushan354//issues)
-
```
Comment (skyv26) [inline LICENSE:100]: - [x] Please make sure that you don't have any broken links in your `README` file because I tried to see your contributing link which is supposed to take me to your repo issue section, but I got a 404 page which means the path that you provided in your `README` file is not correct, kindly fix it.

image

**Comment by bhushan354** [comment]
Hello sir , I hope my project fullfills the criteria but if there is some issue , it would be of great help to me if you tell me on slack or whatsapp +918600118932 , as I have time constraint .

Thank you for consideration,
Bhushan .

File: LICENSE [hunk 3]
```diff
@@ -1,121 +1,7 @@
...
-If you like this project, pplease, feel free to star the repository and share it with others who might benefit from it.
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-## 🙏 Acknowledgments <a name="acknowledgements"></a>
-
-I would like to thank the Microverse team and Code reviewers who helped us create this project. Also, I would like to express our sincere gratitude to [Agustine Soria](https://github.com/SaveryIV) for his valuable assistance in solving a problem with fetching rockets From API. His expertise and guidance Ire instrumental in finding a solution.
```
Comment (skyv26) [inline LICENSE:114]: - [x] Well done, I really appreciate your sincerity in acknowledging your peer/fellow/friend who helped you with the project but as per the requirement, you should acknowledge first the designer of the web application from whom you took the inspiration to design the application. Please follow the requirements carefully and update this section with acknowledgement for the designer of the metrics apps.

image

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @bhushan354, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 

## Required Changes ♻️

- [x] Because your file changes are not reflected in this PR, that's why I am gonna add some comments here. I appreciate your hard work, you did an amazing job by adding your creativity, and design logic and I am happy that instead of 2 you made `three` pages. But you should stick with project requirements, by strongly following the design wireframe, according to which, you should have 2 column design only, and you need to make sure that your application should use only one colour theme, not multiple colours so that's why you need to make some updates to meet the project requirement, your second page I agreed you have two columns but your first page should also have the same design as second. Kudos for the `search` functionality which is working fine.

image

- [x] As I mentioned about colour, you can choose the colour of your own choice but your layout should also use the same colour except for the colours for the text and links. I can see that you 
are using more than 2 colours, so please adhere to the requirements and follow the wireframe colour properly.

image

- [x] Kindly take a look and compare your design, your design looks different in terms of styling, please fix your design styles because adhering to the project requirement will help you to be a better developer and detail-oriented

### Design Comparison
<table>
<tr>
<th>Behance Design</th>
<th>Your Design</th>
</tr>
<tr>
<td>

metrics_app

</td>
<td>

image

</td>
</tr>
</table>

- [x] Please make sure that your application is warm or error-free, sometime Warnings are not that important for us while development, but if you are getting a warning for something really important then you should fix it because as you can see in the below screenshot, you got a warning of missing `attribute` **Key** for the element running inside the loop but because `React` uses the virtual dom and always make the required changes to the real DOM and behind the scenes, this type of keys help the react to differentiate between the things and maintain the uniqueness to perform the render or repaint the component based on our action.

image

- [x] As I mentioned already about your design, I would also take your attention to your detail page, you should make your detail page as per the wireframe provided by Microverse, it seems like you put in your efforts but neglected the important requirement, but I still believe you can do make the changes to pass the requirements.

image

- [x] Always try to separate your test files from your components, you test file should be inside the `__test__` folder just like `component` folders which makes your code base clean and easy to follow.

image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
Hello sir , I hope these changes fullfills the criteria but if there are some issues , it would be of great help to me if you tell me on whatsapp +918600118932 , as I have time constraint .

Thank you for consideration,
Bhushan .

File: README.md
Comment (Whoistolu) [reply README.md:1]: - Well resolved

Comment (Whoistolu) [reply README.md:1]: - [ ] [REQUIRED] Kindly add a professional project description to this project. The previous reviewer only recommended that you add the relevant description to the README not that you should remove it completely. A professional README needs a standard project description.

File: LICENSE (hunk 1 above)
Comment (Whoistolu) [reply LICENSE:45]: - [ ] [REQUIRED] This project is still yet to be deployed as recommended by the previous reviewer and as stated by the project requirement. Kindly deploy your project properly and share the correct deployed link in the README file. Kindly deploy your project properly and add the deployment link here.

File: LICENSE (hunk 2 above)
Comment (Whoistolu) [reply LICENSE:100]: - well resolved

File: LICENSE (hunk 3 above)
Comment (Whoistolu) [reply LICENSE:114]: - well resolved

File: README.md [hunk 4]
```diff
@@ -111,11 +109,11 @@ If you like this project, pplease, feel free to star the repository and share it
...
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-I would like to thank the Microverse team and Code reviewers who helped us create this project. Also, I would like to express our sincere gratitude to [Agustine Soria](https://github.com/SaveryIV) for his valuable assistance in solving a problem with fetching Superheor data From API. His expertise and guidance helped me in finding a solution.
+I would like to thank the Nelson Sakwa from him I took Inspiration to build this project and Microverse team and Code reviewers who helped us create this project. Also, I would like to express our sincere gratitude to [Agustine Soria](https://github.com/SaveryIV) for his valuable assistance in solving a problem with fetching Superheor data From API. His expertise and guidance helped me in finding a solution.
 
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./MIT) licensed.
+This project is [MIT](./LICENSE) licensed.
```
Comment (Whoistolu) [inline README.md:117]: - Well resolved

File: README.md [hunk 5]
```diff
@@ -96,7 +94,7 @@ To get a local copy up and running follow these simple example steps.
 
 Contributions, issues, and feature requests are welcome!
 
-Feel free to check the [issues page](https://github.com/bhushan354//issues)
+Feel free to check the [issues page](https://github.com/bhushan354/Superhero-Search-Project/issues)
```
Comment (Whoistolu) [inline README.md:97]: - Well resolved

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @bhushan354 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
hello sir you have written well resolved in here so what does it mean @Whoistolu

**Comment by Whoistolu** [comment]
> Hello sir , I hope these changes fullfills the criteria but if there are some issues , it would be of great help to me if you tell me on whatsapp +918600118932 , as I have time constraint .
> 
> Thank you for consideration, Bhushan .

You are almost there @bhushan354  🥇 . Kindly check the comments I added for the remaining changes under the comments of the previous reviewer. The project description on the README and the deployment are still pending. Let's fix those 👍🏽

**Comment by bhushan354** [comment]
sir I dont know why the github pages is deploying the README @Whoistolu , can you help me with that , please!!

**Comment by Whoistolu** [comment]
> sir I dont know why the github pages is deploying the README @Whoistolu , can you help me with that , please!!

So sorry about this experience @bhushan354 , kindly ensure that you have selected the correct branch to deploy from. Pease ensure that you are selecting the `feature` branch.

**Comment by bhushan354** [comment]
yes sir I have selected it correctly @Whoistolu

**Comment by Reem-lab** [comment]
Hello @bhushan354 could you check your slack?

**Comment by bhushan354** [comment]
Hello sir/mam , I hope these changes fullfills the criteria but if there are some issues , it would be of great help to me if you tell me on whatsapp +918600118932 , as I have time constraint and have to submit this project in less than 1 hour and my everything else is completed .

Thank you for consideration,
Bhushan .

**Comment by bhushan354** [comment]
> Hello @bhushan354 could you check your slack?

yes wait

**Review by omar-muhamad — COMMENTED** [review]
# Changes Required ♻️
### Hi @,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the required changes [here](https://github.com/bhushan354/Superhero-Search-Project/pull/7#pullrequestreview-1678499081):
As required by the last reviewer, Please add a proper project description here to the readme file.

image

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
